packages/eslint-plugin/tests/rules/no-unnecessary-type-assertion.test.ts

```diff
@@ -416,6 +416,18 @@ enum T {
 const b = 1 as T.Value2;
       `,
     },
+    `
+const foo: unknown = {};
+const bar: object = foo!;
+const baz: {} = foo!;
+    `,
```

Comment:
[Testing] This is really two tests in one: `: object = foo!` and `: {} = foo!`. Could you please split it into two? We prefer granular tests to get more granular feedback on what's failed and make it more clear what specific thing is being tested).